lib/wallaroo/core/barrier/barrier_token.pony

```diff
@@ -48,10 +49,14 @@ primitive InitialBarrierToken is BarrierToken
 class val AutoscaleBarrierToken is BarrierToken
   let _worker: String
   let _id: AutoscaleId
+  let _leaving_workers: Array[WorkerName] val
 
-  new val create(worker': String, id': AutoscaleId) =>
+  new val create(worker': String, id': AutoscaleId,
+    lws: Array[WorkerName] val)
```

Comment:
I prefer `leaving_workers: Array[WorkerName] val` here, as it would be clearer for future us when reading this code.